### DISCUSSION TITLE
Training loss and measure metrics

### DISCUSSION THREAD

**Discussion by wangjiawen2013** [tracel-ai/burn#3174] [Q&A]
Hi,
I have a silly question. We usually use loss (such as crossentropyloss/mseloss) for backpropagation during training, while use recall, precsion, F1-score as performance metrics. Why not try the other way round ? I mean, use recall, precsion, F1-score for backpropagation during training and use loss as performance metrics. Is it technical feasible ?

**Comment by laggui** [comment]
These metrics rely on hard predictions (e.g., thresholding logits to 0 or 1), which are non-differentiable. You can’t compute a gradient of "how many true positives" with respect to the model’s parameters in a smooth way.

There are soft approximations that can be used to better match the metric you would like to actually optimize, but just like these loss functions it is still a surrogate (though better approximations are helpful for the target metric).
